### PR TITLE
partial revert of  "mgr/DaemonServer: log pgmap usage to cluster log" and adds a new 'stats' cluster log channel

### PR DESCRIPTION
**Pull request by jecluis**

### CODE REVIEW

**Comment by jecluis** [comment]
@liewegas @jdurgin is this more acceptable then?

File: src/mgr/MgrStandby.cc [hunk 1]
```diff
@@ -52,6 +52,7 @@ MgrStandby::MgrStandby(int argc, const char **argv) :
   log_client(g_ceph_context, client_messenger.get(), &monc.monmap, LogClient::NO_FLAGS),
   clog(log_client.create_channel(CLOG_CHANNEL_CLUSTER)),
   audit_clog(log_client.create_channel(CLOG_CHANNEL_AUDIT)),
+  stats_clog(log_client.create_channel(CLOG_CHANNEL_STATS)),
```
Comment (liewegas) [inline src/mgr/MgrStandby.cc]: missing from this commit

File: src/mon/Monitor.cc [hunk 2]
```diff
@@ -583,6 +584,14 @@ void Monitor::update_log_clients()
...
     return;
 
+  for (auto x : { clog, audit_clog, stats_clog }) {
+    x->update_config(log_to_monitors, log_to_syslog,
+		     log_channel, log_prio, log_to_graylog,
+		     log_to_graylog_host, log_to_graylog_port,
+		     fsid, host);
+  }
+
+  /*
```
Comment (liewegas) [inline src/mon/Monitor.cc]: can drop teh commented out section?

**Comment by liewegas** [comment]
nits aside, looks good.

shall we move the osdmap print and mgrmap prints to this channel as well?

File: src/mon/Monitor.cc (hunk 2 above)
Comment (jecluis) [reply src/mon/Monitor.cc]: oh man, forgot that. _sigh_

**Comment by jecluis** [comment]
> nits aside, looks good.
> 
> shall we move the osdmap print and mgrmap prints to this channel as well?

I thought about going for a more in-depth changeset, but I then realized that it should probably be handled at a later time; otherwise I think the PR will get bloated - that'll be a funny rabbit hole.

File: src/mgr/MgrStandby.cc (hunk 1 above)
Comment (jecluis) [reply src/mgr/MgrStandby.cc]: I don't think so. For some reason, I think github is just mangling the commit order:


145ca1a27a (gh-rw/wip-clog-log-level, wip-clog-log-level) mgr: log pg stats to 'stats' cluster log channel
00f44c792b mgr: add 'stats' cluster log channel
d801bdb446 mon: add 'stats' cluster log channel
9eeabf3cc9 common/LogEntry: drop check dating back to 0.87
b2aa442e88 Revert "mgr/DaemonServer: log pgmap usage to cluster log"

Comment (jecluis) [reply src/mgr/MgrStandby.cc]: (that, fwiw, is `git log --oneline`, bottom to top is the order)

Comment (liewegas) [reply src/mgr/MgrStandby.cc]: ah, i hate it when github does that!

**Comment by neha-ojha** [comment]
retest this please

**Comment by vumrao** [comment]
https://tracker.ceph.com/issues/39646

**Comment by jdurgin** [comment]
the old decode tests are failing due to the removed check in LogEntry::decode - you can add that type to the ceph-object-corpus intentionally incompatible types to skip testing on older versions

**Comment by jdurgin** [comment]
adding the channel looks good to me!

**Comment by jecluis** [comment]
@jdurgin would a directory within the object-corpus repo with the (commit ver/forward_incompat) + a file with the class name do the trick? or should I go with the already existing 15.0-something directory?

**Comment by jdurgin** [comment]
@jecluis the existing 15.x directory is fine, I think a dir for LogEntry and LogSummary would suffice

**Comment by jecluis** [comment]
uh, I guess MLog gets affected by these changes as well, so it gets broken too...

**Comment by jecluis** [comment]
seems this is good to go, as soon as the ceph-object-corpus changes are merged

**Comment by smithfarm** [comment]
unstale

**Comment by smithfarm** [comment]
@jecluis Is this eligible for backport to octopus and/or nautilus? If so, could you open a tracker?

**Comment by jdurgin** [comment]
@jecluis seems this one is pretty outdate - feel free to reopen if you still want to pursue it

**Pull request closed without merging**